conjur/cli.py

```diff
@@ -16,31 +16,96 @@
 
 from .version import __version__
 
+class ArgparseWrapper(argparse.ArgumentParser):
+    def error(self, message):
+        sys.stderr.write('Error %s\n' % message)
+        self.print_help()
+        sys.exit(1)
+
+    def subcommand_error(self, message, help):
+        sys.stderr.write('Error %s\n' % message)
+        sys.stderr.write('%s\n' % help)
+        sys.exit(1)
+
+    def parse_args(self, args=None, namespace=None):
+        args, argv = self.parse_known_args(args, namespace)
+        action = args.resource if args else None
+        if argv:
+            msg = 'unrecognized arguments: %s'
+            err_msg = (msg % ' '.join(argv))
+            if action:
+                # Find the requesting action
+                action_help = self._subparsers._actions[0].choices.get(action)
+                if action_help:
+                    # Return the help for the specific action
+                    self.subcommand_error(err_msg, action_help.format_help())
+                else:
+                    self.error(err_msg)
+            else:
+                self.error(err_msg)
+        return args
 
 class Cli():
     """
     Main wrapper around CLI-like usages of this module. Provides various
     helpers around parsing of parameters and running client commands.
     """
 
+    @staticmethod
+    def main_description():
+        """
+        This method builds the header for the main screen
+        """
+        return '''
+        
+Conjur’s CLI for managing roles, resources and privileges
+
+Usage:
+    conjur [global options] command [subcommand] [options] [arguments…]'''
+
+    @staticmethod
+    def main_epilog():
+        """
+        This method builds the footer for the main help screen
+        """
+        return '''
+Copyright:
+    Copyright © 1999-2020 CyberArk Software Ltd. All rights reserved.
+    <www.cyberark.com>
+
+    To get help on a specific command, type: conjur <command> -h'''
+
+    @staticmethod
+    def title(title):
+        """
+        This method builds a reusable title for each argument section
+        """
+        return'''
+%s''' %title
```

Comment:
This seems really messy. Why not just use `return "\n{}".format(title)`?

Reply:
Absolutely no reason 🙃 I just thought that Python was space sensitive so I needed to be exact with my spaces and didn't think to use the formatter. Thanks!